lib/utils/queryable.js

```diff
@@ -111,16 +111,14 @@ const getter = ({ data, key }) => {
       }, {})
       return _data
     } else {
-      // if can't find any more values, it means it's just over
-      // and there's nothing to return
-      if (Object.hasOwn(_data, k) && !_data[k]) {
-        _data = ''
-      } else if (!_data[k]) {
-        return undefined
-      } else {
-        // otherwise sets the next value
-        _data = _data[k]
-      }
+        const value = _data[k]
+        if (typeof value === 'boolean' || (value === '' && Object.hasOwn(_data, k))) {
+            _data = value
+        } else if (value === undefined) {
```

Comment:
if value === undefined then returning `_data[k]` will return undefined anyways. Not sure what this check gains us.

Reply:
You're right, I did overcomplicate that. The undefined check is needed since other functionality/tests expect it to be returned and not just set to `_data`.

I simplified it so now it only returns undefined, else just sets up `_data = value`

Reply:
> The undefined check is needed since other functionality/tests expect it to be returned and not just set to _data.

Can you show me where this is happening? That still seems very counterintuitive.

```
if (x = 2) {

Reply:
That works, change has been made